spec/commands/whereami_spec.rb

```diff
@@ -277,4 +278,15 @@ def blimey!
   it "should work inside an object" do
     expect(pry_eval(Object.new, 'whereami')).to match(/Inside #<Object/)
   end
+
+  it 'should work after chdir from main script' do
+    script_dir = File.expand_path(File.join(__dir__, '..', 'fixtures'))
+    script = 'whereami_main_chdir.rb'
+
+    output, status = Open3.capture2(RbConfig.ruby, script, chdir: script_dir)
+
+    expect(status.success?).to be true
+    expect(output.strip).to eq File.join(script_dir, script)
```

Comment:
There are specs failing for Ruby >3.0. [link](https://github.com/pry/pry/actions/runs/8807297811/job/24174148613?pr=2312)

Reply:
Yup. Looks like `target.eval('__dir__')` returns `nil` in versions greater than 2.7. Not sure how to get around that... is there something else that could get the actual pathname or dirname of a file after 3.0?